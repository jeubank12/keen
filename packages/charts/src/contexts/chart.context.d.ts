import React from 'react';
import { Dimension, Margins, Theme, ScaleSettings } from '../types';
export declare type ChartContextType = {
  theme: Theme;
  svgDimensions: Dimension;
  margins: Margins;
  xScaleSettings: ScaleSettings;
  yScaleSettings: ScaleSettings;
  data: object[];
};
declare const ChartContext: React.Context<{}>;
export default ChartContext;
