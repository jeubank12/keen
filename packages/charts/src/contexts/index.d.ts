import ChartContext, { ChartContextType } from './chart.context';
import ScaleContext, { ScaleContextType } from './scale.context';
export { ChartContext, ScaleContext, ScaleContextType, ChartContextType };
