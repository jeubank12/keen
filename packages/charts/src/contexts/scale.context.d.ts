import React from 'react';
import { ScaleBand, ScaleLinear } from 'd3-scale';
export declare type ScaleContextType = {
  xScale: ScaleBand<string> | ScaleLinear<number, number>;
  yScale: ScaleLinear<number, number> | ScaleBand<string>;
};
declare const ScaleContext: React.Context<{}>;
export default ScaleContext;
