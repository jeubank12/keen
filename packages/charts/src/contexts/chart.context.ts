import React from 'react';
import { ScaleSettings } from '@keen.io/charts-utils';

import { Dimension, Margins, Theme } from '../types';

export type ChartContextType = {
  theme: Theme;
  svgDimensions: Dimension;
  margins: Margins;
  xScaleSettings: ScaleSettings;
  yScaleSettings: ScaleSettings;
  data: Record<string, any>[];
};

const ChartContext = React.createContext({});

export default ChartContext;
