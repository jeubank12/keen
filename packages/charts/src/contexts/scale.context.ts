import React from 'react';
import { ScaleBand, ScaleLinear } from 'd3-scale';

export type ScaleContextType = {
  xScale: ScaleBand<string> | ScaleLinear<number, number>;
  yScale: ScaleLinear<number, number> | ScaleBand<string>;
};

const ScaleContext = React.createContext({});

export default ScaleContext;
