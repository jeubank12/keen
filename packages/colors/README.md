# `@keen.io/charts`

Basic color palette for keen.io projects.
