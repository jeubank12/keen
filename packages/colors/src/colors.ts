export const colors = {
  black: {
    100: '#4F5B5F',
    200: '#3A494D',
    300: '#4F5B5F',
    400: '#273437',
    500: '#1D2729',
  },
  white: {
    500: '#FFFFFF',
    400: '#ECF5F7',
    300: '#E1E2E4',
  },
  purple: {
    500: '#674AA9',
    400: '#7153B5',
    300: '#8063C2',
    200: '#9376D5',
    100: '#A689E9',
  },
  orange: {
    500: '#CB5623',
    400: '#D95B24',
    300: '#E66C37',
    200: '#F28252',
    100: '#F69F79',
  },
  yellow: {
    500: '#E29B1E',
    400: '#F0AB30',
    300: '#FBBC57',
    200: '#FFD898',
    100: '#FFE7C2',
  },
  green: {
    500: '#487650',
    400: '#578760',
    300: '#6C9A74',
    200: '#85AB8B',
    100: '#9CC0A2',
  },
  gray: {
    500: '#E1E2E4',
    400: '#D8D8D8',
    300: '#F2F2F2',
  },
  pink: {
    500: '#F4A083',
    400: '#F4AD94',
    300: '#F7BCA7',
    200: '#F6C9B9',
    100: '#FADACE',
  },
  lightBlue: {
    500: '#85B4C3',
    400: '#8FC2D3',
    300: '#A2D3E3',
    200: '#B7E3F1',
    100: '#C9ECF8',
  },
  blue: {
    500: '#27566D',
    400: '#31627A',
    300: '#44748C',
    200: '#5989A2',
    100: '#77A3BB',
  },
};
