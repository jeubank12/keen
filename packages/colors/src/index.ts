export const colors = {
  white: '#fff',
  black: '#1d2729',
  blue: '#27566d',
  green: '#487650',
  orange: '#cb5623',
  violet: '#6042a1',
  yellow: '#e29b1e',
  marine: '#85b4c3',
  pink: '#f4q083',
  gray: '#E1E2E4',
  darkBlue: '#27566D',
};
