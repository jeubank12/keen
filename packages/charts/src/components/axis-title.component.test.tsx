import React from 'react';
import { mount } from 'enzyme';
import { colors } from '@keen.io/colors';

import { ChartContext } from '../contexts';

import AxisTitle from './axis-title.component';

const line = { x1: 0, x2: 100, y1: 100, y2: 0 };
const groupBox = { x: 10, y: 10, height: 10 };
const orientation = 'horizontal';
const title = 'Axis Title';

const setup = (overProps: any = {}) => {
  const props = {
    line,
    groupBox,
    orientation,
    ...overProps,
  };

  const context = {
    theme: {
      axisX: {
        tickSize: 10,
        tickPadding: 10,
        labels: {},
        title: {
          alignment: 'left',
          typography: {
            fontStyle: 'normal',
            fontWeight: 'normal',
            fontSize: 14,
            fontFamily: 'Lato Bold, sans-serif',
            fontColor: colors.blue['500'],
          },
        },
      },
      axisY: {
        tickSize: 10,
        tickPadding: 10,
        labels: {},
        title: {
          alignment: 'top',
          typography: {
            fontStyle: 'normal',
            fontWeight: 'normal',
            fontSize: 10,
            fontFamily: 'Lato Bold, sans-serif',
            fontColor: colors.blue['300'],
          },
        },
      },
    },
  };

  const wrapper = mount(
    <ChartContext.Provider value={context}>
      <svg>
        <AxisTitle {...props}>{title}</AxisTitle>
      </svg>
    </ChartContext.Provider>
  );

  return { wrapper, props };
};

describe('<AxisTitle />', () => {
  it('should render AxisTitle', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should render AxisTitle text', () => {
    const { wrapper } = setup();
    expect(wrapper.find('text').text()).toEqual(title);
  });

  it('should render horizontal title with theme props', () => {
    const { wrapper } = setup();
    expect(wrapper.find('text')).toMatchInlineSnapshot(`
      <text
        fill="#27566D"
        fontFamily="Lato Bold, sans-serif"
        fontSize={14}
        fontStyle="normal"
        fontWeight="normal"
        style={Object {}}
        textAnchor="start"
        x={0}
        y={40}
      >
        Axis Title
      </text>
    `);
  });

  it('should render vertical title with theme props', () => {
    const { wrapper } = setup({ orientation: 'vertical' });
    expect(wrapper.find('text')).toMatchInlineSnapshot(`
      <text
        fill="#44748C"
        fontFamily="Lato Bold, sans-serif"
        fontSize={10}
        fontStyle="normal"
        fontWeight="normal"
        style={
          Object {
            "transform": "rotate(-90deg)",
          }
        }
        textAnchor="end"
        x={-0}
        y={-10}
      >
        Axis Title
      </text>
    `);
  });
});
