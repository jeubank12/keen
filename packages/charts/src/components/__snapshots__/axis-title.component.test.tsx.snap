// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AxisTitle /> should render AxisTitle 1`] = `
<svg>
  <AxisTitle
    groupBox={
      Object {
        "height": 10,
        "x": 10,
        "y": 10,
      }
    }
    line={
      Object {
        "x1": 0,
        "x2": 100,
        "y1": 100,
        "y2": 0,
      }
    }
    orientation="horizontal"
  >
    <text
      fill="#27566D"
      fontFamily="Lato Bold, sans-serif"
      fontSize={14}
      fontStyle="normal"
      fontWeight="normal"
      style={Object {}}
      textAnchor="start"
      x={0}
      y={40}
    >
      Axis Title
    </text>
  </AxisTitle>
</svg>
`;
